tests/unit/client.py

```diff
@@ -247,10 +247,41 @@ def avoid_network_race_condition():
     db_file = db_dir+'/haas.db'
     conn = sqlite3.connect(db_file)
     cur = conn.cursor()
+    timeout = 0
     while True:
         que = len(cur.execute('select * from networking_action;').fetchall())
         if que > 0:
             time.sleep(1)
+            timeout = timeout + 1
+            if timeout > 10:  # breakout if exceeds 10 secs.
+                break
+        else:
+            break
+
+
+def avoid_race_sqlalchemy():
+    from haas.model import Node, Project, NetworkingAction
+    from sqlalchemy import func
+    from sqlalchemy import create_engine
+    from sqlalchemy.orm import sessionmaker
+
+    global db_dir  # Value of db_dir is populated by `create_setup`.
+    uri = 'sqlite:///'+db_dir+'/haas.db'
+    engine = create_engine(uri)
+    Session = sessionmaker(bind=engine)
+    session = Session()
```

Comment:
You could cut this down a bit by just using db.session, and a `with app.app_context():` None of the session setup, besides setting the URI in the config. It also gets rid of most of the imports above.